src/Microsoft.AspNet.Routing/RouteBuilder.cs

```diff
@@ -21,6 +23,14 @@ public RouteBuilder(IApplicationBuilder applicationBuilder, IRouter defaultHandl
                 throw new ArgumentNullException(nameof(applicationBuilder));
             }
 
+            if (applicationBuilder.ApplicationServices.GetService(typeof(RoutingMarkerService)) == null)
+            {
+                throw new InvalidOperationException(Resources.FormatUnableToFindServices(
```

Comment:
Forgot to ask last time: Do we really want the constructor to `throw` for a situation unrelated to its parameters? Realize `RouteBuilder` objects aren't passed into the system but am still hoping there's another place where we can cover the non-`UseRouting()` scenario.